src/routers/axelar/XCMRouter.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.18;
+pragma abicoder v2;
+
+struct Multilocation {
+    uint8 parents;
+    bytes[] interior;
+}
+
+// https://github.com/PureStake/moonbeam/blob/v0.30.0/precompiles/xcm-transactor/src/v2/XcmTransactorV2.sol#L12
+interface XcmTransactorV2 {
+    function transactThroughSignedMultilocation(
+        Multilocation memory dest,
+        Multilocation memory feeLocation,
+        uint64 transactRequiredWeightAtMost,
+        bytes memory call,
+        uint256 feeAmount,
+        uint64 overallWeight
+    ) external;
+}
+
+struct XcmWeightInfo {
+    // The weight limit in Weight units we accept amount to pay for the
+    // execution of the whole XCM on the Centrifuge chain. This should be
+    // transactWeightAtMost + an extra amount to cover for the other
+    // instructions in the XCM message.
+    uint64 buyExecutionWeightLimit;
+    // The weight limit in Weight units we accept paying for having the Transact
+    // call be executed. This is the cost associated with executing the handle call
+    // in the Centrifuge.
+    uint64 transactWeightAtMost;
+    // The amount to cover for the fees. It will be used in XCM to buy
+    // execution and thus have credit for pay those fees.
+    uint256 feeAmount;
+}
+
+interface AxelarExecutableLike {
+    function execute(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) external;
+}
+
+interface AxelarGatewayLike {
+    function callContract(string calldata destinationChain, string calldata contractAddress, bytes calldata payload)
+        external;
+}
+
+contract ConnectorAxelarXCMRouter is AxelarExecutableLike {
+    address constant XCM_TRANSACTOR_V2_ADDRESS = 0x000000000000000000000000000000000000080D;
+
+    mapping(address => uint256) public wards;
```

Comment:
let's add a todo to inherit from Auth directly. 
I already added the contract to the repo in my pr, so we can fix once it is merged